core/domain/fs_domain.py

```diff
@@ -108,96 +107,105 @@ def version(self):
+    story/story_id/assets/).An asset has no meaning outside its entity, so the
+    assets in these asset folders should therefore not be edited directly. They
+    should only be modified as side-effects of some other operation (such as
+    adding an image to an entity like topics, stories or explorations).
 
     In general, assets should be retrieved only within the context of the
-    exploration that contains them, and should not be retrieved outside this
+    entity that contains them, and should not be retrieved outside this
     context.
 
     Args:
-        exploration_id: str. The id of the exploration.
+        entity_name: str. The name of the entity (eg: exploration, topic etc).
+        entity_id: str. The ID of the corresponding entity.
     """
 
     _DEFAULT_VERSION_NUMBER = 1
 
-    def __init__(self, exploration_id):
-        """Constructs a ExplorationFileSystem object.
+    def __init__(self, entity_name, entity_id):
+        """Constructs a EntityFileSystem object.
 
         Args:
-            exploration_id: str. The id of the exploration.
+        entity_name: str. The name of the entity (eg: exploration, topic etc).
+        entity_id: str or None. The ID of the corresponding entity. None is to
+            accommodate tests in some files and some older exp jobs where a
+            single exp_id is used to initialize class.
         """
-        self._exploration_id = exploration_id
+        if entity_id is None:
+            self._assets_path = entity_name
+        else:
+            self._assets_path = entity_name + '/' + entity_id
 
     @property
-    def exploration_id(self):
-        """Returns the id of the exploration.
+    def assets_path(self):
+        """Returns the path of the parent folder of assets.
 
         Returns:
-            str. The exploration id.
+            str. The path.
         """
-        return self._exploration_id
+        return self._assets_path
 
     def _get_file_metadata(self, filepath, version):
         """Return the desired file metadata.
 
         Returns None if the file does not exist.
 
         Args:
-            filepath: str. The path to the relevant file within the exploration.
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
             version: int. The version number of the file whose metadata is to be
                 returned.
 
         Returns:
             FileMetadataModel or None. The model instance representing the file
-                metadata with the given exploration_id, filepath, and version,
+                metadata with the given assets_path, filepath, and version,
                 or None if the file does not exist.
         """
         if version is None:
             return file_models.FileMetadataModel.get_model(
-                self._exploration_id, 'assets/%s' % filepath)
+                self._assets_path, 'assets/%s' % filepath)
```

Comment:
This reads a little weirdly because we're combining _assets_path with a string starting with 'assets'.

Do you think it would be better to include assets/ in the _assets_path as well? We talk about the "entity's assets folder" and to me that implies blah/blah/assets rather than blah/blah.

Reply:
Yeah, as long as there is no other folder in the {{exp_id}} folder, which I think is the case. (Then, why not get rid of the assets folder altogether and have images, audio etc. in {{exp_id}} itself?)

Reply:
That seems reasonable too, but it would involve another migration, so I think it's fine to stick with the current convention :)

Reply:
Yeah, done.